x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -51,6 +52,56 @@ export type IndexPatternPrivateState = IndexPatternPersistedState & {
   indexPatterns: Record<string, IndexPattern>;
 };
 
+export function IndexPatternDataPanel(props: DatasourceDataPanelProps<IndexPatternPrivateState>) {
+  return (
+    <div>
+      Index Pattern Data Source
+      <div>
+        <EuiComboBox
+          data-test-subj="indexPattern-switcher"
+          options={Object.keys(props.state.indexPatterns).map(key => ({
```

Comment:
nit: Could be shortened by `Object.entries(props.state.indexPatterns).map(({ title, id }) => ({`

Reply:
I think you mean `Object.values`, but yes that works.